ignite/services/network/publish.go

```diff
@@ -175,25 +217,25 @@ func (n Network) Publish(ctx context.Context, c Chain, options ...PublishOption)
 
 	msgs := []sdk.Msg{msgCreateChain}
 
-	if !o.shares.Empty() {
+	if len(o.sharePercentages) != 0 {
 		totalSharesResp, err := n.campaignQuery.TotalShares(ctx, &campaigntypes.QueryTotalSharesRequest{})
 		if err != nil {
 			return 0, 0, 0, err
 		}
 
 		var coins []sdk.Coin
 
-		for _, share := range o.shares {
-			amount := int64((float64(share.Amount.Int64()) / 100) * float64(totalSharesResp.TotalShares))
-			coins = append(coins, sdk.NewInt64Coin(share.Denom, amount))
+		for _, share := range o.sharePercentages {
+			amount := int64(share.percent * float64(totalSharesResp.TotalShares/100))
```

Comment:
can you please use `sdk.Dec` instead of `float64` so we cannot lose the precision. The `sdk.Dec` is the default for cosmos-sdk and wraps a big number inside

Reply:
Didn't used sdk.Dec, but now there are no floats